app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -244,9 +244,8 @@ protected void deliverResult( LoginActivity receiver, Integer result) {
                         case DataPullTask.DOWNLOAD_SUCCESS:
                             if(!tryLocalLogin(true)) {
                                 receiver.raiseLoginMessage(StockMessages.Auth_CredentialMismatch, true);
-                            } else {
-                                break;
                             }
+                            break;

```

Comment:
Pretty sure we always want to break here, not just when the local login succeeds. Otherwise when the login fails it will show two error messages: bad auth credentials and bad network connect.